astropy/io/fits/connect.py

```diff
@@ -94,6 +112,9 @@ def read_table_fits(input, hdu=None):
         The HDU to read the table from.
     """
 
+    # Avoid circular imports
+    from .time import FitsTime
```

Comment:
I will cross check once more. I initially had done some tests and thinking over this but then again I may have overlooked somethings. So I'll check!